### PR TITLE
Change the priority of HeaderCheck to error.

### LINKED ISSUE #128: Change the priority of HeaderCheck from info to error
> kaikreuzer 16 hours ago Owner

>Hm, I found out why: Because it is considered prio 3.
@svilenvul Did we do that by intention? I would actually hope that a missing header breaks the build - and where we have exceptions because we need different headers, it would be nice if we could deactivate this check in the pom of the appropriate bundle.

See https://github.com/openhab/openhab2-addons/pull/1711#discussion_r131113839

### PR DESCRIPTION
**Pull request by svilenvul**
Closes #128

Signed-off-by: Svilen Valkanov <svilen.valkanov@musala.com>

### CODE REVIEW

**Comment by martinvw** [comment]
Did you check the current openhab2 repo, will it survive?

**Comment by svilenvul** [comment]
Surprisingly, it survived :)

File: src/main/resources/rulesets/checkstyle/rules.xml [hunk 1]
```diff
@@ -28,12 +28,12 @@
         -error   -->
     <!-- The default level of a check is "error". -->
     <!-- See http://checkstyle.sourceforge.net/config.html#Severity -->
-    <property name="severity" value="info" />
-    <property name="ignoreLines" value="1,2,3"/>
+    <property name="severity" value="error" />
+    <property name="ignoreLines" value="2"/>
```
Comment (martinvw) [inline src/main/resources/rulesets/checkstyle/rules.xml]: which line is line 2?

Comment (svilenvul) [inline src/main/resources/rulesets/checkstyle/rules.xml]: The line containing the year, unfortunately it is different in the openhab-core and openhab2-addons repo, so no idea how to overcome this for now ...

Comment (martinvw) [inline src/main/resources/rulesets/checkstyle/rules.xml]: @kaikreuzer wdyt? Not really the best line to ignore imho.

File: src/main/resources/rulesets/checkstyle/rules.xml [hunk 2]
```diff
@@ -28,12 +28,12 @@
...
     <property name="fileExtensions" value="java"/>
     <property name="header"
         value="${checkstyle.headerCheck.content}"
-        default="/**\n * Copyright (c) 2010-year by the respective copyright holders.\n *\n * All rights reserved. This program and the accompanying materials\n * are made available under the terms of the Eclipse Public License v1.0\n * which accompanies this distribution, and is available at\n * http://www.eclipse.org/legal/epl-v10.html\n */"/>  </module>
+        default="/**\n * Copyright (c) 2010-2017 by the respective copyright holders.\n *\n * All rights reserved. This program and the accompanying materials\n * are made available under the terms of the Eclipse Public License v1.0\n * which accompanies this distribution, and is available at\n * http://www.eclipse.org/legal/epl-v10.html\n */"/>  </module>
```
Comment (martinvw) [inline src/main/resources/rulesets/checkstyle/rules.xml]: Can't we make the end year dynamic? Maybe by string replacing it `${current_year}` => _some java code_

Comment (svilenvul) [inline src/main/resources/rulesets/checkstyle/rules.xml]: I guess we should move to [RegexpHeaderCheck](http://api.dpml.net/checkstyle/3.5/com/puppycrawl/tools/checkstyle/checks/RegexpHeaderCheck.html).

**Pull request merged**